.github/workflows/release.yml

```diff
@@ -1,175 +1,109 @@
-      with:
-        name: mdbook-svgbob-Windows
-        path: artifacts/
-
-    - name: Extract the version tag
-      id: version_tag
-      # or ${GITHUB_REF##*/}
-      shell: bash
-      run: echo ::set-output name=value::$(echo $GITHUB_REF | cut -d / -f 3)
-
-    - name: Create Release
-      id: create_release
-      uses: actions/create-release@v1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        tag_name: ${{ github.ref }}
-        release_name: ${{ github.ref }}
-        draft: true
-        prerelease: true
-
-    - name: Upload Release Asset (linux)
-      uses: actions/upload-release-asset@v1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include an `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
-        upload_url: ${{ steps.create_release.outputs.upload_url }}
-        asset_path: artifacts/mdbook-svgbob-Linux.zip
-        asset_name: mdbook-svgbob-${{ steps.version_tag.outputs.value }}-linux.zip
-        asset_content_type: application/zip
-
-    - name: Upload Release Asset (mac)
-      uses: actions/upload-release-asset@v1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        upload_url: ${{ steps.create_release.outputs.upload_url }}
-        asset_path: artifacts/mdbook-svgbob-macOS.zip
-        asset_name: mdbook-svgbob-${{ steps.version_tag.outputs.value }}-macos.zip
-        asset_content_type: application/zip
-
-    - name: Upload Release Asset (win)
-      uses: actions/upload-release-asset@v1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        upload_url: ${{ steps.create_release.outputs.upload_url }}
-        asset_path: artifacts/mdbook-svgbob-Windows.zip
-        asset_name: mdbook-svgbob-${{ steps.version_tag.outputs.value }}-windows.zip
-        asset_content_type: application/zip
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Get Artifacts
+        uses: actions/download-artifact@v3
+      - name: Create Release
+        uses: softprops/action-gh-release@v1
+        with:
+          files: mdbook-svgbob-*/mdbook-svgbob-*
 
   publish:
-    name: publish
+    name: Publish to crates.io
     needs: release
     runs-on: ubuntu-latest
-
-    strategy:
-      fail-fast: true
-
     steps:
-    - name: Checkout
-      uses: actions/checkout@v3
-
-    - name: Publish crate
-      uses: actions-rs/cargo@v1.0.1
-      env:
-        CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
-      with:
-        toolchain: stable
-        command: publish
-        # args: --token ${{ secrets.CARGO_REGISTRY_TOKEN }}
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Rust
+        uses: dtolnay/rust-toolchain@stable
```

Comment:
Just a small comment from a random bystander: the runners for GitHub actions already come with the latest Rust stable release.

See [here](https://github.com/actions/runner-images/commit/d54fa255120bfd0f508c1feb53490190787b1316#diff-d3f496e99457b699df2a5621c8f59cb52f60a050e1a88318af7d6f7069c12b2eL153-L157) for an example of how the runner image was updated with Rust 1.65. That version was released on November 3rd and the image was updated on November 24. So it's probably not necessary to do anything special when a recent stable release is good enough.

Reply:
Thanks for sharing this! Somehow I wasn't aware of that. I think for the scope of this project a always recent stable rust version should be fine.